vaadin-maps-leaflet-flow-demo/src/main/java/software/xdev/vaadin/maps/leaflet/flow/demo/LeafletView.java

```diff
@@ -119,72 +122,397 @@ private void initMapComponents()
+			},
+			imageOverlay::remove
+		));
+	}
+	
+	private void addVideoOverlayDemo()
+	{
+		final LVideoOverlay videoOverlay = new LVideoOverlay(
+			this.reg,
+			"https://www.mapbox.com/bites/00188/patricia_nasa.webm",
+			new LLatLngBounds(
+				this.reg,
+				new LLatLng(this.reg, 32, -130),
+				new LLatLng(this.reg, 13, -100)
+			),
+			new LVideoOverlayOptions().withAttribution(
+				"<a href='https://www.youtube.com/watch?v=ZcxBOQTOb2Q'>NASA</a>"));
+		this.hlButtons.add(this.createToggleButton(
+			"Show video",
+			"Hide video",
+			() -> {
+				videoOverlay.addTo(this.map);
+				// Start video 30s in (otherwise no hurricane - boring)
+				videoOverlay.invokeSelf(".getElement().currentTime=30");
+				
+				this.map.setView(new LLatLng(this.reg, 20, -115), 5);
+				
+				// Play otherwise stops after hiding
+				videoOverlay.invokeSelf(".getElement()?.play()");
+			},
+			videoOverlay::remove
+		));
+	}
+	
+	private void addComplexPolygonDemo()
+	{
+		// See also 38.871008516069516, -77.0559701487416
+		final LPolygon pentagon = new LPolygon(
+			this.reg,
+			List.of(
+				List.of(
+					List.of(
+						new LLatLng(this.reg, 38.872923, -77.054687),
+						new LLatLng(this.reg, 38.870639, -77.053252),
+						new LLatLng(this.reg, 38.868895, -77.055569),
+						new LLatLng(this.reg, 38.870071, -77.058473),
+						new LLatLng(this.reg, 38.872570, -77.057877)
+					),
+					List.of(
+						new LLatLng(this.reg, 38.871749, -77.055494),
+						new LLatLng(this.reg, 38.870866, -77.054932),
+						new LLatLng(this.reg, 38.870225, -77.055829),
+						new LLatLng(this.reg, 38.870650, -77.056901),
+						new LLatLng(this.reg, 38.871607, -77.056723)
+					)
+				),
+				List.of(
+					new LLatLng(this.reg, 38.871107, -77.055903),
+					new LLatLng(this.reg, 38.870990, -77.055831),
+					new LLatLng(this.reg, 38.870899, -77.055950),
+					new LLatLng(this.reg, 38.870961, -77.056104),
+					new LLatLng(this.reg, 38.871089, -77.056072)
+				)
+			));
+		final String imageLink =
+			"https://upload.wikimedia.org/wikipedia/commons/thumb/f/f9/Building_and_ship_comparison_to_the_Pentagon2"
+				+ ".svg/260px-Building_and_ship_comparison_to_the_Pentagon2.svg.png";
+		pentagon.bindPopup("<a href='https://en.wikipedia.org/wiki/The_Pentagon' target='_blank'>"
+			+ "<center><b>The Pentagon</b></center><br>"
+			+ "<img style='width:12em' src='" + imageLink + "'>"
+			+ "</a>");
+		this.hlButtons.add(this.createToggleButton(
+			"Show complex polygon",
+			"Hide complex polygon",
+			() -> {
+				pentagon.addTo(this.map);
+				pentagon.openPopup();
+				
+				this.map.setView(new LLatLng(this.reg, 38.871008, -77.055970), 17);
+			},
+			pentagon::remove
+		));
+	}
+	
+	// S5413 - Yes it's used correctly
+	// S2245 - This is a reproducible demo
+	// S1215 - This is a memory test and we don't rely on random GC collects
+	@SuppressWarnings({"java:S5413", "java:S2245", "java:S1215", "java:S3776"})
+	private void addMemoryTestDemo()
```

Comment:
This is such a strange test.
What's its purpose?
Could it be simplified?

Reply:
> What's its purpose?

To test if the memory on the client side is freed up correctly when handling client side components.
This is a rather extreme test (thousands of components are added and removed) but it needs to be that extreme as otherwise you don't notice the RAM changes in the browser.

Added a bit of documentation and renamed the button hopefully that should be clearer now.

> Could it be simplified?

I doubt that.

Reply:
Why should the user check if the memory is freed up correctly? This is not a use case but something that should be tested by the developers of the library (us). The demo should display some use cases, meaning for what and how the library can be used.
I think this should be replaced by some simple function that shows what you can do with the LComponentManagementRegistry instead. This would be interesting.

Reply:
> Why should the user check if the memory is freed up correctly? This is not a use case but something that should be tested by the developers of the library (us). The demo should display some use cases, meaning for what and how the library can be used.

IMHO this is also kind of a extreme usecase/benchmark when adding thousands of markers to a map. 
It demonstrates how our library handles this without causing resource problems. 
I also see no other way in how we can test this (Selenium is no option as we can't how browsers allocate memory or how they clean it up).

> I think this should be replaced by some simple function that shows what you can do with the LComponentManagementRegistry instead.

Normally you don't need to use ``LComponentManagementRegistry`` methods manually:
* ``writeOptions...``, ``add`` and ``clientComponentJsAccessor`` are framework internal used for Leaftlet component methods or initialization
* ``remove``, ``freeUpClient`` and ``clearAll`` are only needed when working with huge amounts of components and/or freeing up memory manually

So I think this "Benchmark" is already a good demonstration.